1.0.0/fix/code cleanning

JulesdeCube opened the pull request:
# Desciption
clean repository structure, auto format code , comment and header gard

# Change

🔀MERGE:'version/1.0.0' to get last feature 

**FIX:**
- `Makefile`:
  -🩹 remove `-e` flag on echo
  -🐛 move tmp folder creation before package installing
- 🎨`src/data_set/*`: change the gard and file path
- 🎨`src/preprocessing/*`: change the gard and file path
- 🎨`src/segmentation/*`: change the gard and file path
- 🎨`src/neural_network/neural_network.h`: add file desciptions
- 🎨src/preprocessing/resize_char.h`: change format of the gard

**MODIFY:**
- 🧰`Makefile`: split `SRC` and `DEPS` into sub module
- 📝`README.md`: add the orc feature and the programe require
- 🚚`src/main.c`: change include path
- 🚚`src/segmentation/auto_rotation.h`: change include path

**RENAME:**
- 🚚`src/images/*` -> `src/data_set`: match module structure
- 🚚`src/load_Image/*` -> `src/preprocessing/*` : match module name

**REMOVE:**
- 🔥`src/utlis/bool.h`: never import
- 🔥`test/save_file`: trash file

JulesdeCube committed:
🧰MODIFY: Makefile splitting

**FIX:**
- `Makefile`:
  -🩹 remove `-e` flag on echo
  -🐛 move tmp folder creation before package installing

**MODIFY:**
- 🧰`Makefile`: split `SRC` and `DEPS` into sub module

JulesdeCube committed:
📝ADD: Readme require and feature

**MODIFY:**
- 📝`README.md`: add the orc feature and the programe require

JulesdeCube committed:
🚚RENAME: change data set module place

**MODIFY:**
- 🧰 `Makefile`: change deprency place
- 🚚`src/main.c`: change include path

**RENAME:**
- 🚚`src/images/*` -> `src/data_set`: match module structure

JulesdeCube committed:
🔥REMOVE: unused files

**REMOVE:**
- 🔥`src/utlis/bool.h`: never import
- 🔥`test/save_file`: trash file

JulesdeCube committed:
🎨MODIFY: auto formating source files

JulesdeCube committed:
🚚RENAME: change segmentation module place

**MODIFY:**
- 🧰 `Makefile`: change deprency place
- 🚚`src/main.c`: change include path

**RENAME:**
- 🚚`src/load_Image/auto_roation*` -> `src/segmentation/auto_roation*`:
  match module structure
- 🚚`src/load_Image/segmentation*` -> `src/segmentation/segmentation*`:
  match module structure

JulesdeCube committed:
🚚RENAME: preprocessing folder to `preprocessing`

**MODIFY:**
- 🧰 `Makefile`: change deprency place
- 🚚`src/main.c`: change include path
- 🚚`src/segmentation/auto_rotation.h`: change include path

**RENAME:**
- 🚚`src/load_Image/*` -> `src/preprocessing/*` : match module name

JulesdeCube committed:
🎨FIX: normalize all the header gard

**FIX:**
-🎨 `src/data_set/*`: change the gard and file path
-🎨 `src/preprocessing/*`: change the gard and file path
-🎨 `src/segmentation/*`: change the gard and file path
-🎨 `src/neural_network/neural_network.h`: add file desciptions

JulesdeCube committed:
🔀MERGE:'version/1.0.0' to get last feature

JulesdeCube committed:
🎨FIX: `resize_char.h` gard

**FIX:**
- 🎨`src/preprocessing/resize_char.h`: change format of the gard

Winbay commented on src/preprocessing/tools.h:
```diff
@@ -11,24 +11,25 @@
 
 /**
- ** \brief Search the withest (maximum) and the blackest (minimum) pixel of a pixbuf 
+ ** \brief Search the withest (maximum) and the blackest (minimum) pixel of a pixbuf
```
witest => whitest

JulesdeCube committed:
🎨FIX: doxygen comment

**FIX:**
- 🎨`src/preprocessing/*.h`: remove space before doxygen comment.

Winbay commented on src/preprocessing/resize_char.c:
```diff
@@ -0,0 +1,62 @@
+
+#define SIZE_OF_IMAGE_CHAR 28
+
+double *get_great_char(GtkWidget *image)
```
double *image_to_array(GtkWidget *image)

Winbay commented on src/preprocessing/load.c:
```diff
@@ -0,0 +1,54 @@
+    //remove_noise_image(imagec, 1);
+    //remove_noise_image(imagec, 2);
+
+    double *array = get_great_char(imagec);
```
double *array = image_to_array(imagec);

JulesdeCube committed:
🐛FIX: `get_great_char` to `image_to_array`

**FIX:**
- 🐛`src/preprocessing/load.c` : rename to `image_to_array`
- 🐛`src/preprocessing/resize_char.c` : rename to `image_to_array`

**MODIFY:**
- 📝`src/preprocessing/tool.c` : otrtograph correction

JulesdeCube committed:
🔀MERGE: resolve conflict 'version/1.0.0'

JulesdeCube committed:
🔥REMOVE: trash file from merge

**REMOVE:**
- 🔥`src/load_Image/resize_char.c`: trash
- 🔥`src/load_Image/resize_char.h`: trash
